docs/reference/builder.md

```diff
@@ -1329,7 +1329,7 @@ defined and the `what_user` value was passed on the command line. Prior to its d
 
 > **Warning:** It is not recommended to use build-time variables for
 >  passing secrets like github keys, user credentials etc. Build-time variable
->  values are visible to any user of the image with the `docker history` command.
+>  values may be visible to any user of the image with the `docker history` command.
```

Comment:
Is this actually un-knowable? Things like "may be" inject uncertainty "Will they be visible? Won't they? How to know?" Can we explain when they would or wouldn't? Maybe we can just link to the section below? And instead of "may be" change it to "are visible if you specify them as `ARG` lines in the Dockerfile?